test/test_train_mp_imagenet.py

```diff
@@ -132,13 +151,30 @@ def train_imagenet():
 
   torch.manual_seed(42)
 
+  devices = (
+      xm.get_xla_supported_devices(
+          max_devices=FLAGS.num_cores) if FLAGS.num_cores != 0 else [])
   device = xm.xla_device()
   model = get_model_property('model_fn')().to(device)
+  writer = SummaryWriter(log_dir=FLAGS.logdir) if FLAGS.logdir else None
   optimizer = optim.SGD(
       model.parameters(),
       lr=FLAGS.lr,
       momentum=FLAGS.momentum,
       weight_decay=5e-4)
+  total_num_devices = len(
```

Comment:
You need to use `xm.get_world_size()` to get the total number of devices.
Even the non MP one, will not work correctly on POD.

Reply:
let me know if it looks right now